datafusion-examples/examples/advanced_udf.rs

```diff
@@ -191,6 +199,48 @@ impl ScalarUDFImpl for PowUdf {
     }
 }
 
+/// Evaluate `base ^ exp` *without* allocating a new array, if possible
+fn pow_in_place(base: f64, exp_array: ArrayRef) -> Result<ArrayRef> {
```

Comment:
Here is the example of how to write a function that reuses the input argument's allocation